### PR TITLE
Optionally specifiy custom host with `-p`

### PR DESCRIPTION
**Pull request by metachris**
**What current issue(s) does this address, or what feature is it adding?**

This PR allows `prompt.py` users to specify custom hosts with the `-p` privatenet argument. Useful for instance for events such as hackathons, where people want to connect to the privnet running on a shared server somewhere.

Help output:

    $ python prompt.py -h
    usage: prompt.py [-h] [-m | -p [host] | --coznet | -c CONFIG]
                    [-t {dark,light}] [-v] [--version]

    optional arguments:
    -h, --help            show this help message and exit
    -m, --mainnet         Use MainNet instead of the default TestNet
    -p [host], --privnet [host]
                            Use a private net instead of the default TestNet,
                            optionally using a custom host (default: 127.0.0.1)
    --coznet              Use the CoZ network instead of the default TestNet
    -c CONFIG, --config CONFIG
                            Use a specific config file
    -t {dark,light}, --set-default-theme {dark,light}
                            Set the default theme to be loaded from the config
                            file. Default: 'dark'
    -v, --verbose         Show smart-contract events by default
    --version             show program's version number and exit

Normal privatenet behavior (as before):

    $ python prompt.py -p

With a custom host:

    $ python prompt.py -p example.com
    Using custom privatenet host: example.com
    - Seeds: example.com:20333, example.com:20334, example.com:20335, example.com:20336
    - RPC: http://example.com:30333

With an invalid host:

    $ python prompt.py -p http://example.com
    Traceback (most recent call last):
    File "prompt.py", line 1048, in <module>
        main()
    File "prompt.py", line 1004, in main
        settings.setup_privnet(args.privnet)
    File "/Users/chris/Projects/chris/neo/neo-python/neo/Settings.py", line 161, in setup_privnet
        raise Exception("No protocol prefix or port allowed in host, use just the IP or domain.")
    Exception: No protocol prefix or port allowed in host, use just the IP or domain.
    
**How did you solve this problem?**

Added nargs to argparse line in prompt, and added a `host` argument in `Settings.setup_privnet`

**How did you make sure your solution works?**

Manual tests

**Are there any special changes in the code that we should be aware of?**

No

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [X] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: neo/Settings.py [hunk 1]
```diff
@@ -147,9 +147,23 @@ def setup_testnet(self):
...
 
-    def setup_privnet(self):
-        """ Load settings from the privnet JSON config file """
+    def setup_privnet(self, host=None):
+        """
+        Load settings from the privnet JSON config file
+
+        Args:
+            host (string): can be either an IP which will be used for seeds with standard ports and RPC
+                           or in the format of `IP:port`, which will be used as the only seed
```
Comment (brianlenz) [inline neo/Settings.py]: How would you use this `IP:port` format? It looks like colons aren't supported below, but I could be missing something...

Comment (metachris) [reply neo/Settings.py]: Oh i left that comment in. The problem I faced with port was that you'd need 2 ports: at least a p2p and a rpc server port. Didn't have an idea about how to do this properly, so i thought let's start simple and improve later. If you have any ideas i'd be happy to hear them! ;)

Comment (ixje) [inline neo/Settings.py]: While I don't think you'll need more than 1 host on a privatenet (and thus this solution would suffice), one thing I'd like to mention just for consideration is the option to specify a `list` of 'host:port' pairs and replace the whole SEED_LIST or RPC_LIST with that. 

I am struggling a bit with the thought that this PR sort of solves the same problems as the `-c` option does, no?

Comment (brianlenz) [reply neo/Settings.py]: That is a good point. Perhaps for the hackathon "shared" private net use case, they should just be distributing a json file for everyone to use with `-c`?

Comment (metachris) [reply neo/Settings.py]: It just makes it easier. For instance a hackathon, you spin up a VM somewhere with privnet + neoscan + faucet. Then you just tell people to run `./prompt.py -p hackathon.cityofzion.io` and that's it. No need to copy around protocol-config json files beforehand.

True, a single host should be enough.

The question is how we'd define the RPC port, but at the same time keep things as simple as possible.

Comment (metachris) [reply neo/Settings.py]: At hackathons, every thing you don't need to do for the system setup is worth it's weight in gold. Just make it as easy as possible for participants, and I think this is it.

Comment (brianlenz) [reply neo/Settings.py]: Cool, you definitely know better than me on this one 😉 I'm good with keeping it simple, but I think we probably only need to support the single host scenario. For anything more complex, I would think you should just be using a config file.

Comment (metachris) [reply neo/Settings.py]: This is a great middle ground.

**Comment by metachris** [comment]
Thanks so much for the brainstorming guys!

**Comment by localhuman** [comment]
Looks great!

**Pull request merged**